commercial/app/controllers/BookOffersController.scala

```diff
@@ -20,46 +17,35 @@ class BookOffersController(bookFinder: BookFinder, bestsellersAgent: Bestsellers
   with implicits.Collections
   with implicits.Requests {
 
-  private def booksSample(isbns: Seq[String], segment: Segment): Future[Seq[Book]] =
-    bestsellersAgent.getSpecificBooks(isbns) map { specificBooks =>
-      (specificBooks ++ bestsellersAgent.bestsellersTargetedAt(segment)).distinctBy(_.isbn).take(4)
-    }
+  private def booksSample(isbns: Seq[String], segment: Segment): Seq[Book] =
+    (bestsellersAgent.getSpecificBooks(isbns) ++ bestsellersAgent.bestsellersTargetedAt(segment)).distinctBy(_.isbn).take(4)
+
+  private def isValidIsbn(isbn: String): Boolean = (isbn forall (_.isDigit)) && (isbn.length == 10 || isbn.length == 13)
```

Comment:
>Are all those parenthesis necessary?
If I'm not using dot notation then I believe so, yes - the `&&` appears to bind more closely than the `forall` when not using dot notation.

> I would stick with the dot notation to call methods and not mix dot and space in the same statement.

I generally agree with that sentiment but in this case I am only using dot notation on instance methods i.e. the `String` instance method of `isDigit`. I do not think that it can be called in the method you suggest above: `isbn.forall(isDigit)` as `isDigit` is an instance method, not a class/object method.

Therefore I think I'm being consistent in using space notation in chaining operations and the `_.` notation when invoking instance methods. I know that `post-fix` operations can be imported and then used but this just seems to muddy the waters in my opinion, hence I've kept it as `isbn.length` and not `isbn length`.